runtime/exelib/common/memcheck.c

```diff
@@ -753,6 +753,9 @@ static void *memoryCheck_allocate_memory32(OMRPortLibrary *portLib, UDATA byteAm
 	memCheckPortLib->tty_printf( memCheckPortLib, "allocate_memory32(%d)\n", byteAmount);
 #endif
 
+#if defined(OMR_ENV_DATA64)
+	portLib->port_copy_suballocator_heap_globals(memCheckPortLib, portLib);
+#endif /* defined(OMR_ENV_DATA64) */
```

Comment:
Why shouldn't this go in `memoryCheck_initialize()` instead?

Reply:
You're right. I was just thinking this too. It doesn't need to be called every time we allocate.

Reply:
Actually. I remember why I did it this way. The original `portLib` `PPG` options are not yet set when `memCheck_initialize` is called; that is why I pushed it back to the allocator wrapper.

Reply:
I still don't think here is the right place to propagate those options. It should happen just once in the life-cycle of memcheck.

Reply:
I agree. I'm looking into exactly where a more appropriate placement would be.

Reply:
If we are to change where we set up the `PPG` options for the `memCheckPortLib`, the call to the set up routine would need to be [here](https://github.com/eclipse-openj9/openj9/blob/028a24e1522017110a37a883cd9c1de10f9c7291/runtime/vm/jvminit.c#L7512).

The preceding condition calls `runInitializationStage(vm, ALL_LIBRARIES_LOADED)` which sets up the `PPG` options in the source `portLib`. Within the call in the next line, `J9RelocateRASData(vm);`, there is an allocation through the suballocator. When `-Xcheck` is used, it uses the `memCheckPortLib` sub-allocator wrapper.

We could add a function pointer on the `portLib` that would be assigned in `memoryCheck_initialize` to a procedure defined in `memcheck.c` that calls `port_copy_suballocator_heap_globals`. Then we'd call that function in the location where I've linked to above.

I'd prefer to leave things as they are now (since the setup is contained within the `memcheck.c`), instead of an approach similar to the one that I've described.

@keithc-ca let me know what you think, or if you had any other ideas.

Reply:
@keithc-ca I'd like to get this merged. Could you give some input on what I have said above?